app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java

```diff
@@ -292,12 +293,13 @@ private void setFieldsFromSharedPreferences(NavigationRoute.Builder builder) {
 
   private Locale getLocale() {
     SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
-    return new Locale(sharedPreferences.getString("language", ""));
+    String localeString = sharedPreferences.getString("language", "device_locale");
```

Comment:
In order to increase readability, what about replacing magic numbers (literals in general including `String`s) with constants with a particular meaning?
Further information: [Replace Magic Number with Symbolic Constant](https://refactoring.com/catalog/replaceMagicNumberWithSymbolicConstant.html).